keps/sig-storage/1495-volume-populators/README.md

```diff
@@ -573,6 +583,12 @@ resource usage (CPU, RAM, disk, IO, ...) in any components?**
   is determined, a populator-specific investigation will be needed, starting
   from looking at the logs for that populator.
 
+## Drawbacks
+
+  `DataSource` and `DataSourceRef` cannot be pointed to different resources.
+  While this is by design, it does not solve certain use case like restore volume 
+  from VolumeSnapshot, and then do volume population.
```

Comment:
I don't quite understand this use case.  Can you elaborate?

Reply:
In the VP KEP design: If `DataSource` is set valid AND `DataSourceRef` is set differently; reject

Which means DataSource and DataSourceRef **cannot be different**. The use case I described: customer sets snapshot in the DataSource field, creating a volume from the snapshot. And then do volume population via volume populator.

This case is currently not supported.

Reply:
In that case, the volume population will overwrite the volume created from snapshot?

Reply:
I think the dataSource field has higher priority than the dataSourceRef field according to the provisioner code: https://github.com/kubernetes-csi/external-provisioner/blob/master/pkg/controller/controller.go#L1946

It sets DataSource first if it exists and then it will ignore the DataSourceRef field.

Reply:
But it does not change the fact that VP right now cannot do restoring a volume from snapshot first from dataSource and then do the population from dataSourceRef.

But I think you can do restore from snapshot and populatio from a new CRD in dataSourceRef.